File: MonSingleton.java
public class MonSingleton {
 private static MonSingleton monSingle;
 private static int compteur;
 private static int nbAppel;

 private MonSingleton() {
  compteur = 1;
  nbAppel = 0;
 }

 public static MonSingleton obtientInstance() {
  if (monSingle == null) {
   monSingle = new MonSingleton();
  }
  nbAppel++;
  return monSingle;
 }

 public void augmente() {
  compteur++;
 }

 public String toString() {
  return "Compteur =  " + compteur + " (" + nbAppel + ")";
 }

 public static void main(String[] args) {
  MonSingleton ms1 = MonSingleton.obtientInstance();
  ms1.augmente();
  MonSingleton ms2 = MonSingleton.obtientInstance();
  ms2.augmente();
  System.out.println(ms1);
  System.out.println(ms2);
 }
}

File: MonSingleton.cpp
#include <iostream>
#include <sstream>
#include <string>

using namespace std;

class MonSingleton
{
private:
 /* data */
 static MonSingleton monSingleton;
 static int nbAppel;
 int compteur;

 MonSingleton(int le_compteur) : compteur(le_compteur)
 {
  nbAppel = 0;
 }
 MonSingleton(MonSingleton &);
 void operator=(MonSingleton &);

public:
 static MonSingleton &obtientInstance()
 {
  nbAppel++;
  return monSingleton;
 }
 void augmente()
 {
  compteur++;
 }
 string toString()
 {
  ostringstream sortie;
  sortie << "Compteur = ";
  sortie << compteur << "(" << nbAppel << ")" << ends;

  return sortie.str();
 }
};

MonSingleton MonSingleton::monSingleton(1);
int MonSingleton::nbAppel(1);

int main(int argc, char const *argv[])
{
 MonSingleton &ms1 = MonSingleton::obtientInstance();
 ms1.augmente();

 MonSingleton &ms2 = MonSingleton::obtientInstance();
 ms2.augmente();

 cout << ms1.toString() << endl;
 return 0;
}
